### PR TITLE
Bugfix MTE-1054 [v116] Re-enable Performance Tests

### PR DESCRIPTION
**Pull request by jjSDET**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-TODO)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/TODO)

### Description
This update changes the names of the test db fixtures for re-enabling performance tests for `Bookmarks` and `History`.

The tests have updated workflows to isolate the performance measurements.

### Pull requests checks where applicable
- [x] Fill in the three TODOs above (tickets number and description of your work)
- [x] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [ ] Unit tests written and passing
- [ ] Documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by isabelrios** [comment]
While I run the test, there are some changes that we need (not sure if you are addressing these as part of this PR.. though)
- Only datafixture tests should be enabled in the Fennec_Enterprise_XCUITests scheme
Screenshot 2023-06-28 at 12 01 13
- Move the perf tests to the specific test suite so that they run as part of the perf job

File: Tests/XCUITests/DatabaseFixtureTest.swift [hunk 1]
```diff
@@ -32,73 +32,34 @@ class DatabaseFixtureTest: BaseTestCase {
...
-            // activity measurement here
-            app.launch()
-        }
+        waitForExpectations(timeout: TIMEOUT, handler: nil)
+        let historyList = app.tables["History List"].cells.count
+        XCTAssertEqual(historyList, 101, "There should be 101 entries in the history list")
     }
 
     func testPerfBookmarks1000openMenu() {
+        waitForTabsButton()
```
Comment (isabelrios) [inline Tests/XCUITests/DatabaseFixtureTest.swift:62]: Remove this test from here, it has been moved to the TabsPerformance test suite

File: Tests/XCUITests/TabsPerformanceTests.swift [hunk 2]
```diff
@@ -5,16 +5,15 @@
...
 
     override func setUp() {
         // Test name looks like: "[Class testFunc]", parse out function name
         let parts = name.replacingOccurrences(of: "]", with: "").split(separator: " ")
         let functionName = String(parts[1])
         let archiveName = fixtures[functionName]
-
         // defaults
-        launchArguments = [LaunchArguments.PerformanceTest, LaunchArguments.SkipIntro, LaunchArguments.SkipWhatsNew, LaunchArguments.SkipETPCoverSheet, LaunchArguments.SkipContextualHints]
+        launchArguments = [LaunchArguments.PerformanceTest, LaunchArguments.SkipIntro, LaunchArguments.SkipWhatsNew, LaunchArguments.SkipETPCoverSheet, LaunchArguments.SkipContextualHints, LaunchArguments.LoadDatabasePrefix + fixtures[functionName]!]
```
Comment (isabelrios) [inline Tests/XCUITests/TabsPerformanceTests.swift:16]: Adding the database here does not work, if you look at the logs the testAppStarted crashes. That is because is not in the fixtures list. That's why we have the if condition below in L19.

File: Tests/XCUITests/TabsPerformanceTests.swift [hunk 3]
```diff
@@ -133,4 +139,65 @@ class TabsPerformanceTest: BaseTestCase {
...
+            XCTCPUMetric(), // to measure cpu cycles
+            XCTStorageMetric(), // to measure storage consuming
+            XCTMemoryMetric()]) {
+            // activity measurement here
+            app.launch()
+            waitForTabsButton()
+        }
+    }
+
+    func testPerfBookmarks1000openMenu() {
```
Comment (isabelrios) [inline Tests/XCUITests/TabsPerformanceTests.swift:188]: This test works locally for me but fails on Bitrise:
```
testPerfBookmarks1000openMenu, Asynchronous wait failed: Exceeded timeout of 45 seconds, with unfulfilled expectations: "Expect predicate `count == 1001` for object <XCUIElementQuery: 0x600001124140>".
```
We may need to increase the timeout...

File: Tests/XCUITests/TabsPerformanceTests.swift [hunk 4]
```diff
@@ -5,20 +5,26 @@
 import XCTest
 
 class TabsPerformanceTest: BaseTestCase {
-    let fixtures = ["testPerfTabs_1_20startup": "tabsState20.archive", "testPerfTabs_3_20tabTray": "tabsState20.archive", "testPerfTabs_2_1280startup": "tabsState1280.archive", "testPerfTabs_4_1280tabTray": "tabsState1280.archive"]
+    let fixtures = ["testPerfTabs_1_20startup": "tabsState20.archive",
```
Comment (isabelrios) [inline Tests/XCUITests/TabsPerformanceTests.swift:8]: We should rename this test suite since it has more tests now not only those related to tabs

File: Tests/XCUITests/TabsPerformanceTests.swift (hunk 3 above)
Comment (isabelrios) [reply Tests/XCUITests/TabsPerformanceTests.swift:188]: looking at the screenshots, the bookmark list is populated, but for some reason the count of items is not available within that time

File: Tests/XCUITests/DatabaseFixtureTest.swift (hunk 1 above)
Comment (jjSDET) [reply Tests/XCUITests/DatabaseFixtureTest.swift]: done

File: Tests/XCUITests/TabsPerformanceTests.swift (hunk 2 above)
Comment (jjSDET) [reply Tests/XCUITests/TabsPerformanceTests.swift]: ah, i see. i thought i was seeing issues running it locally and the changes fixed it, but it must have been dumb luck.

File: Tests/XCUITests/TabsPerformanceTests.swift (hunk 4 above)
Comment (jjSDET) [reply Tests/XCUITests/TabsPerformanceTests.swift:8]: done

File: bitrise.yml [hunk 5]
```diff
@@ -1475,13 +1475,13 @@ meta:
 
 trigger_map:
 - push_branch: main
-  pipeline: pipeline_multiple_shards
+  workflow: perfherder-test
```
Comment (isabelrios) [inline bitrise.yml:1478]: I think we can undo this change, tests are passing and the PERFHERDER_DATA object looks fine,  @jjSDET is there a way to confirm it is correct?

File: Tests/XCUITests/PerformanceTests.swift [hunk 6]
```diff
@@ -133,4 +139,69 @@ class TabsPerformanceTest: BaseTestCase {
...
+            XCTClockMetric(), // to measure timeClock Mon
+            XCTCPUMetric(), // to measure cpu cycles
+            XCTStorageMetric(), // to measure storage consuming
+            XCTMemoryMetric()]) {
+            // activity measurement here
+                navigator.goto(LibraryPanel_Bookmarks)
+                waitForExistence(app.tables["Bookmarks List"], timeout: TIMEOUT_LONG)
+                let number_cells = app.tables["Bookmarks List"].cells.count
+                // debug line - remove
+                print("Number cells \(number_cells)")
```
Comment (isabelrios) [inline Tests/XCUITests/PerformanceTests.swift:201]: Let's remove this line and the previous one as those were just for testing purposes

**Review by isabelrios — APPROVED** [review]
Changes lok good to me, let's remove the debug / test lines and merge this 🚀

File: bitrise.yml (hunk 5 above)
Comment (jjSDET) [reply bitrise.yml:1478]: Screenshot 2023-06-29 at 11 26 42 AM

Comment (jjSDET) [reply bitrise.yml:1478]: for comparison, you can see all the data is formatted the same as with the previous perfherder data

File: Tests/XCUITests/PerformanceTests.swift (hunk 6 above)
Comment (jjSDET) [reply Tests/XCUITests/PerformanceTests.swift:201]: done

File: Client.xcodeproj/xcshareddata/xcschemes/Fennec_Enterprise_XCUITests.xcscheme [hunk 7]
```diff
@@ -167,7 +167,7 @@
       </BuildActionEntries>
    </BuildAction>
    <TestAction
-      buildConfiguration = "Fennec"
+      buildConfiguration = "Fennec_Enterprise"
```
Comment (isabelrios) [inline Client.xcodeproj/xcshareddata/xcschemes/Fennec_Enterprise_XCUITests.xcscheme:170]: Not sure why this change appears here but it should not be this Fennec_Enterprise configuration for Fennec_Enterprise_XCUITests... please rever this change

File: Client.xcodeproj/xcshareddata/xcschemes/Fennec.xcscheme [hunk 8]
```diff
@@ -140,8 +140,8 @@
    </BuildAction>
    <TestAction
       buildConfiguration = "Fennec"
-      selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
-      selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
+      selectedDebuggerIdentifier = ""
+      selectedLauncherIdentifier = "Xcode.IDEFoundation.Launcher.PosixSpawn"
```
Comment (isabelrios) [inline Client.xcodeproj/xcshareddata/xcschemes/Fennec.xcscheme:144]: are these changes coming after the hack session with @clarmso ? I'm not sure about those, would prefer a confirmation for her who worked with Orla on this to prevent the simulator black screen issue

**Comment by isabelrios** [comment]
Please disable the tests for ALL the smoketest...when the name has been changed, it may be find as a new test suite and so it is enabled by default for all schemes.. see for example for Smoketest2:
Screenshot 2023-06-30 at 09 48 48
